testslide/lib.py

```diff
@@ -271,6 +272,8 @@ def _wrap_signature_and_type_validation(
 
     skip_first_arg = _skip_first_arg(template, attr_name)
 
+    # Add this so docstrings and method name are not altered by the mock
+    @wraps(callable_template)
```

Comment:
Hum... I do understand the idea here, though I'm weighting if letting it explicit "this is a mock", with new name and docstring would be better than "implicit" and keeping the old names... Not strongly opinionated either way though.

Reply:
The idea here is that I would like to prevent as much as possible to have
```
<function _wrap_signature_and_type_validation.<locals>.with_sig_and_type_validation at 0x10ee343b0>
```
as information of the wrapped function.

Doing so makes debugging very very hard (at least I had a particularly hard time to identify the mocked function in `mock_callable_testslide.py` due to the function name).

With this change the function would look like
```
<function Client.sync_function at 0x102386ef0>
```

I can see your point of preserving the information of "this is a mock", in this case we might want to have something like
```
<function TestSldeValidation(_wrap_signature_and_type_validation) at 0x105c17560>
```

If you might like this idea I can update the PR